### PR TITLE
Add msgpack serializer

### PR DESCRIPTION
**Pull request by cnx-tcsikos**
I've added [msgpack v3.3.0](https://github.com/msgpack/msgpack-c/tree/cpp_master) serializer to your benchmark.
Also I had trouble with cross-compiling, so added some cmake changes as well.

I don't have native ubuntu 18.04 at the moment I only did benchmark to windows and aarch64 linux (with docker-for-windows so its not very representative).

### CODE REVIEW

**Comment by cnx-tcsikos** [comment]
Here is the benchmark for Windows VS2019: (50 monster,  300000 samples)

|serializer |configuration|encoded size|encode time|decode time|
|-----------|-------------|------------|-----------|-----------|
|handwritten|unsafe       |10413 B     |1720,5 ms  |2009 ms    |
|handwritten|general      |10413 B     |1667 ms    |2278 ms    |
|bitsery    |fixed buffer |6913 B      |1787 ms    |2663 ms    |
|bitsery    |unsafe read  |6913 B      |2479 ms    |1990,5 ms  |
|yas        |general      |10463 B     |2614 ms    |2046 ms    |
|zpp        |general      |8413 B      |2610 ms    |2476,5 ms  |
|bitsery    |general      |6913 B      |2815,5 ms  |2861 ms    |
|bitsery    |compatibility|7113 B      |2747 ms    |2960,5 ms  |
|bitsery    |brief syntax |6913 B      |2718,5 ms  |3013 ms    |
|yas        |compression  |7315 B      |3614,5 ms  |3626,5 ms  |
|bitsery    |compression  |4213 B      |4208 ms    |4971,5 ms  |
|bitsery    |stream       |6913 B      |3136 ms    |10164 ms   |
|flatbuffers|general      |14924 B     |17920,5 ms |4081 ms    |
|cereal     |general      |10413 B     |13285,5 ms |11454 ms   |
|protobuf   |arena        |10018 B     |12109,5 ms |16140,5 ms |
|msgpack    |general      |8857 B      |6595,5 ms  |23449 ms   |
|yas        |stream       |10463 B     |33767,5 ms |19747,5 ms |
|iostream   |general      |8413 B      |35359,5 ms |25550,5 ms |
|protobuf   |general      |10018 B     |36934 ms   |39598 ms   |

File: boost/CMakeLists.txt [hunk 1]
```diff
@@ -6,7 +6,8 @@ if( UNIX )
     set( boost_CMDbootstrap bootstrap.sh )
     set( boost_CMDb2 b2 )
 else()
-    message(FATAL_ERROR "boost building implemented only on unix")
+    message(WARNING "boost building implemented only on unix")
```
Comment (fraillt) [inline boost/CMakeLists.txt:9]: Could you elaborate, why you don't like FATAL_ERROR here?

File: cereal/CMakeLists.txt [hunk 2]
```diff
@@ -8,6 +8,8 @@ ExternalProject_Add(
         CONFIGURE_COMMAND ""
         BUILD_COMMAND ""
         INSTALL_COMMAND ""
+        CMAKE_ARGS -DCMAKE_CXX_STANDARD=${CMAKE_CXX_STANDARD} -DCMAKE_C_COMPILER='${CMAKE_C_COMPILER}' -DCMAKE_CXX_COMPILER='${CMAKE_CXX_COMPILER}' -DCMAKE_SYSTEM_PROCESSOR=${DCMAKE_SYSTEM_PROCESSOR}
```
Comment (fraillt) [inline cereal/CMakeLists.txt:11]: So this makes it possible to pass arguments to all projects when building from the root folder.
Am I right?

File: protobuf/CMakeLists.txt [hunk 3]
```diff
@@ -6,15 +6,17 @@ ExternalProject_Add(
         URL "https://github.com/protocolbuffers/protobuf/releases/download/v3.11.2/protobuf-cpp-3.11.2.tar.gz"
         URL_MD5 "351a0f5c4edd3f45035417f09133ce73"
         CONFIGURE_COMMAND ${CMAKE_COMMAND} -DCMAKE_INSTALL_PREFIX=${protobuf_PREFIX}
-            -DCMAKE_BUILD_TYPE=Release -Dprotobuf_BUILD_TESTS=OFF
+            -DCMAKE_BUILD_TYPE=Release -Dprotobuf_BUILD_TESTS=OFF -Dprotobuf_MSVC_STATIC_RUNTIME=OFF
```
Comment (fraillt) [inline protobuf/CMakeLists.txt]: I guess I cannot, accept it, as this will not work on ubuntu I guess...

**Review by fraillt — CHANGES_REQUESTED** [review]
Thanks for PR,
And for additional improvements as well.
Could you please make this PR onto develop branch?
I want to test it myself and probably update some libraries in the process before merging to master.

File: boost/CMakeLists.txt (hunk 1 above)
Comment (cnx-tcsikos) [reply boost/CMakeLists.txt:9]: FATAL_ERROR makes the build stop.
On Windows I just want the user to be informed that boost will not be used.

File: cereal/CMakeLists.txt (hunk 2 above)
Comment (cnx-tcsikos) [reply cereal/CMakeLists.txt:11]: yep

File: protobuf/CMakeLists.txt (hunk 3 above)
Comment (cnx-tcsikos) [reply protobuf/CMakeLists.txt]: okay, I'll change it to be platform dependent
Done

**Comment by fraillt** [comment]
Thank you!

**Pull request merged**